Open-Rival/src/BuildingRenderer.cpp

```diff
@@ -0,0 +1,142 @@
+#include "pch.h"
+#include "BuildingRenderer.h"
+
+#include <gl/glew.h>
+#include <map>
+#include <vector>
+
+#include "Palette.h"
+#include "RenderUtils.h"
+#include "Rival.h"
+#include "Shaders.h"
+
+namespace Rival {
+
+    BuildingRenderer::BuildingRenderer(
+        std::map<BuildingType, Spritesheet>& buildingSprites,
+        Texture& paletteTexture):
+        buildingSprites(buildingSprites),
+        paletteTexture(paletteTexture) {
+    }
+
+    SpriteRenderable& BuildingRenderer::getOrCreateRenderable(const Building& building) {
+        
+        int id = building.getId();
+
+        if (renderables.find(id) == renderables.end()) {
+            // No Renderable exists for this Unit; create one
+            const BuildingType type = building.getType();
+            const Spritesheet& sprite = buildingSprites.at(type);
+            renderables.emplace(std::piecewise_construct,
+                                std::forward_as_tuple(id),
+                                std::forward_as_tuple(sprite, 1));
```

Comment:
```
void BuildingRenderer::render(
...
            // If Building is deleted, remove the associated Renderable
            if (building.isDeleted()) {
                renderables.erase(building.getId());
                continue;
            }
```
Doesn't this take care of that?

Reply:
That was the intention, but I don't think this will actually work, because deleted Units / Buildings will most likely be removed by the logic before `render` gets called.

So, by the time we pass the list of Units / Buildings to the renderer, the deleted ones will not be present in that list, which means the entries will never be removed from `renderables`.

Reply:
I was not sure if this exact -implemented- approach would work too, but lets keep this in mind and return back to this when we have a way of testing. Updated TODO.md